From 079d45fa50aedc61785a451a95455ea8e2cf12ff Mon Sep 17 00:00:00 2001
Message-Id: <079d45fa50aedc61785a451a95455ea8e2cf12ff.1507326318.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Fri, 6 Oct 2017 17:44:05 -0400
Subject: [PATCH 1/1] Fix CentOS mistake thinking that Xen is always enabled

We just use the existing CONFIG definition to hook the
code compilation out.

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 arch/x86/kernel/smpboot.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/kernel/smpboot.c b/arch/x86/kernel/smpboot.c
index 0e65ae4..420d2d8 100644
--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -338,6 +338,7 @@ static void __init smp_init_package_map(struct cpuinfo_x86 *c, unsigned int cpu)
 	 */
 	max_physical_pkg_id = DIV_ROUND_UP(MAX_LOCAL_APIC, ncpus);
 
+#ifdef CONFIG_XEN_PVHVM
 	if (x86_hyper == &x86_hyper_xen_hvm) {
 		/*
 		 * RHEL-only. Each logical package has not more than
@@ -348,6 +349,7 @@ static void __init smp_init_package_map(struct cpuinfo_x86 *c, unsigned int cpu)
 		 */
 		__max_logical_packages = min(max_physical_pkg_id, total_cpus);
 	}
+#endif
 
 	size = max_physical_pkg_id * sizeof(unsigned int);
 	physical_to_logical_pkg = kmalloc(size, GFP_KERNEL);
-- 
1.8.3.1

